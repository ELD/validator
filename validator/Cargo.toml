[package]
name = "validator"
version = "0.10.1"
authors = ["Vincent Prouillet <prouillet.vincent@gmail.com"]
license = "MIT"
description = "Common validation functions (email, url, length, ...) and trait - to be used with `validator_derive`"
homepage = "https://github.com/Keats/validator"
repository = "https://github.com/Keats/validator"
keywords = ["validation", "api", "validator"]
edition = "2018"

[dependencies]
url = "2"
regex = "1"
lazy_static = "1"
idna = "0.2"
serde = "1"
serde_derive = "1"
serde_json = "1"
validator_types = { version = "0.10", path = "../validator_types" }
validator_derive = { version = "0.10", path = "../validator_derive", optional = true }
card-validate = { version = "2.2", optional = true }
phonenumber = { version = "0.2", optional = true }
unic-ucd-common = { version = "0.9", optional = true }

[features]
phone = ["phonenumber"]
card = ["card-validate"]
unic = ["unic-ucd-common"]
derive = ["validator_derive"]