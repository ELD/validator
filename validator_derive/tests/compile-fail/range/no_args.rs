use validator::Validate;

#[derive(Validate)]
//~^ ERROR: proc-macro derive panicked
//~^^ HELP: Invalid attribute #[validate] on field `s`: Validator `range` requires at least 1 argument out of `min` and `max`
struct Test {
    #[validate(range())]
    s: i32,
}

fn main() {}
