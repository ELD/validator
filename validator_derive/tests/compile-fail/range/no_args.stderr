error: proc-macro derive panicked
 --> $DIR/no_args.rs:3:10
  |
3 | #[derive(Validate)]
  |          ^^^^^^^^
  |
  = help: message: Invalid attribute #[validate] on field `s`: Validator `range` requires at least 1 argument out of `min` and `max`
